tests/test_single_topology_stereo.py

```diff
@@ -0,0 +1,486 @@
+        """
+  Mrv2202 05192218063D
+
+  5  4  0  0  0  0            999 V2000
+   -0.0541    0.5427   -0.3433 C   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4368    0.0213    0.3859 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.9636    0.0925   -0.4646 H   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4652    0.3942   -1.2109 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.0541    2.0827   -0.3433 Cl  0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  1  3  1  0  0  0  0
+  1  4  1  0  0  0  0
+  1  5  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+
+    # first test: 4 core mapping
+    core = np.array([[0, 0], [1, 1], [2, 2], [3, 3]])
+    s_top = SingleTopologyV2(mol_a, mol_b, core, ff)
+
+    x_a = utils.get_romol_conf(mol_a)
+    x_b = utils.get_romol_conf(mol_b)
+    x0 = s_top.combine_confs(x_a, x_b)
+
+    # check initial chirality
+    vol_a = measure_chiral_volume(x0[0], x0[1], x0[2], x0[3])
+    vol_d = measure_chiral_volume(x0[0], x0[1], x0[2], x0[4])
+    assert vol_a > 0 and vol_d < 0
+
+    # G3_PYRAMIDAL -> G4_TETRAHEDRAL
+    idxs_and_params = s_top.generate_end_state_mol_a()
+    frames = simulate_idxs_and_params(idxs_and_params, x0)
+
+    num_vol_a_pos = 0
+    num_vol_a_neg = 0
+
+    for f in frames:
+        vol_a = measure_chiral_volume(f[0], f[1], f[2], f[3])
+        vol_d = measure_chiral_volume(f[0], f[1], f[2], f[4])
+        if vol_a < 0:
+            num_vol_a_neg += 1
+        else:
+            num_vol_a_pos += 1
+        assert vol_d < 0
+
+    # should be within 5% of 50/50
+    assert abs(num_vol_a_pos / len(frames) - 0.5) < 0.05
+
+    # G3_PYRAMIDAL -> G3_PYRAMIDAL (no dummy atoms)
+    idxs_and_params = s_top.generate_end_state_mol_b()
+    frames = simulate_idxs_and_params(idxs_and_params, x0)
+
+    for f in frames:
+        vol_a = measure_chiral_volume(f[0], f[1], f[2], f[3])
+        vol_d = measure_chiral_volume(f[0], f[1], f[2], f[4])
+        assert vol_a > 0 and vol_d < 0
+
+    # # second test: 3 core mapping
+    core = np.array([[0, 0], [1, 1], [2, 2]])
+    s_top = SingleTopologyV2(mol_a, mol_b, core, ff)
+
+    x_a = utils.get_romol_conf(mol_a)
+    x_b = utils.get_romol_conf(mol_b)
+    x0 = s_top.combine_confs(x_a, x_b)
+
+    # check initial chirality
+    vol_a = measure_chiral_volume(x0[0], x0[1], x0[2], x0[3])
+    vol_d = measure_chiral_volume(x0[0], x0[1], x0[2], x0[5])
+    assert vol_a > 0 and vol_d < 0
+
+    # G2_KINK -> G4_TETRAHEDRAL
+    idxs_and_params = s_top.generate_end_state_mol_a()
+    frames = simulate_idxs_and_params(idxs_and_params, x0)
+
+    num_vol_a_pos = 0
+    num_vol_a_neg = 0
+
+    for f in frames:
+        vol_a = measure_chiral_volume(f[0], f[1], f[2], f[3])
+        vol_d = measure_chiral_volume(f[0], f[1], f[2], f[5])
+        if vol_a < 0:
+            num_vol_a_neg += 1
+        else:
+            num_vol_a_pos += 1
+        assert vol_d < 0
```

Comment:
turn this into a function, since it's repeated a few times

Reply:
hm I'm not sure the best way to functionalize this - the idxs that go into `measure_chiral_volume` can be 8 different idxs, not all tests permit pos and neg vols for a single atom. Was it the `if` statement itself that seemed verbose?

Reply:
just that the calculation of the fraction positive seemed a bit repetitive in the 3 examples here. Up to you if it doesn't seem easy to functionalize that's fine too.